python/lsst/cp/pipe/defects.py

```diff
@@ -0,0 +1,645 @@
+        """
+        expCopy = exp.clone()
+        del exp  # for safety - no longer needed as we have a copy
+        maskedIm = expCopy.maskedImage
+
+        defects.maskPixels(expCopy.maskedImage, "BAD")
+        detector = expCopy.getDetector()
+
+        disp = afwDisplay.Display(0, reopenPlot=True, dpi=200)  # , dpi=200)
+
+        if imageType == "flat":  # set each amp image to have a mean of 1.00
+            for amp in detector:
+                ampIm = maskedIm.image[amp.getBBox()]
+                ampIm -= afwMath.makeStatistics(ampIm, afwMath.MEANCLIP).getValue() + 1
+
+        mpDict = maskedIm.mask.getMaskPlaneDict()
+        for plane in mpDict.keys():
+            if plane in ['BAD']:
+                continue
+            disp.setMaskPlaneColor(plane, afwDisplay.IGNORE)
+
+        # if self._getNumGoodPixels(maskedIm) >= 1:
+        disp.scale('asinh', 'zscale')
+        disp.setMaskTransparency(80)
+        disp.setMaskPlaneColor("BAD", afwDisplay.RED)
+
+        disp.setImageColormap('gray')
+        title = f"Detector: {detector.getName()[-3:]} {detector.getSerial()}"
+        title += f", Type: {imageType}, visit: {visit}"
+        disp.mtv(maskedIm, title=title)
+
+        cgUtils.overlayCcdBoxes(detector, isTrimmed=True, display=disp)
+
+    def _plotAmpHistogram(self, dataRef, exp, visit, nSigmaUsed):  # pragma: no cover
+        """
+        Make a histogram of the distribution of pixel values for each amp.
+
+        Main image data histogram plotted in blue. Edge pixels, if masked, are
+        in red. Note that masked edge pixels do not contribute to the
+        underflow and overflow numbers.
+
+        Parameters
+        ----------
+        pdfPages: `matplotlib.backends.backend_pdf.PdfPages`
+            If not None, append per-amp histograms to this object
+        """
+        fig, ax = plt.subplots(nrows=4, ncols=4, sharex='col', sharey='row', figsize=(13, 10))
+
+        detector = exp.getDetector()
+        expTime = exp.getInfo().getVisitInfo().getExposureTime()
+
+        for (amp, a) in zip(reversed(detector), ax.flatten()):
+            mi = exp.maskedImage[amp.getBBox()]
+
+            # normalise by expTime as we plot in ADU/s and don't always work with master calibs
+            mi.image.array /= expTime
+            stats = afwMath.makeStatistics(mi, afwMath.MEANCLIP | afwMath.STDEVCLIP)
+            mean, sigma = stats.getValue(afwMath.MEANCLIP), stats.getValue(afwMath.STDEVCLIP)
+
+            # Get array of pixels
+            EDGEBIT = exp.maskedImage.mask.getPlaneBitMask("EDGE")
+            imgData = mi.image.array[(mi.mask.array & EDGEBIT) == 0].flatten()
+            edgeData = mi.image.array[(mi.mask.array & EDGEBIT) != 0].flatten()
+
+            thrUpper = mean + nSigmaUsed*sigma
+            thrLower = mean - nSigmaUsed*sigma
+
+            nRight = len(imgData[imgData > thrUpper])
+            nLeft = len(imgData[imgData < thrLower])
+
+            nsig = 6
```

Comment:
Should you be using the nSigma numbers from the config here to define the min/max ranges instead?

Reply:
Not exactly, in that it needs to be larger than them, but I take your point about them being hard coded not being good. I'll change it to be the config option + 1.2 so that the default it still 6, but it adjusts dynamically when that option changes at runtime. Thanks for that 👍